.github/workflows/shared-publish-to-nuget-versioned.yaml

```diff
@@ -0,0 +1,112 @@
+on:
+  workflow_call:
+    inputs:
+      release_type:
+        description: The type of version number to return. Must be one of [Patch, Minor or Major]
+        required: true
+        type: string
+      vulnerability_failure_severity:
+        description: The severity to fail the workflow if such vulnerability is detected. DO NOT override it unless a Jira ticket is raised. Must be one of ['CRITICAL', 'CRITICAL,HIGH' or 'CRITICAL,HIGH,MEDIUM'] (without space in between).
+        type: string
+        default: 'CRITICAL,HIGH'
+      dotnet_version:
+        type: string
+        default: 5.0
+      working_dir:
+        description: The path to the directory for which the version should be determined.
+        type: string
+        default: '.'
+
+env:
+  REPO: ${{ github.event.repository.name }}
+
+jobs:
+  release:
+    name: Create Release
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: write
+      security-events: write
+      packages: write
+    steps:
+      - name: Show Context
+        run: |
+          printenv
+          echo "$GITHUB_CONTEXT"
+        shell: bash
+        env: 
+            GITHUB_CONTEXT: ${{ toJson(github) }}
+
+      - name: Check branch and release type
+        id: checkRelease
+        uses: IABTechLab/uid2-shared-actions/actions/check_branch_and_release_type@v2
+        with:
+          release_type: ${{ inputs.release_type }}
+    
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup dotnet ${{ inputs.dotnet_version }}
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ inputs.dotnet_version }}
+
+      - name: Vulnerability Scan
+        uses: IABTechLab/uid2-shared-actions/actions/vulnerability_scan_filesystem@v2
+        with:
+          scan_severity: HIGH,CRITICAL
+          failure_severity: ${{ inputs.vulnerability_failure_severity }}
+
+      - name: Set version number
+        id: version
+        uses: IABTechLab/uid2-shared-actions/actions/version_number@kcc-UID2-2329-add-dotnet-pipeline
+        with:
+          type: ${{ inputs.release_type }}
+          branch_name: ${{ github.ref }}
+          working_dir: ${{ inputs.working_dir }}
+          short_name: Yes
+
+      - name: Update UID2.Client.nuspec
+        run: |
+          current_version=$(grep -o '<version>.*</version>' ${{ inputs.working_dir }}/UID2.Client.nuspec | head -1 | sed 's/<version>\(.*\)<\/version>/\1/')
+          simple_version=${{ steps.version.outputs.simple_version }} 
+          sed -i "s/$current_version/$simple_version/g" ${{ inputs.working_dir }}/UID2.Client.nuspec
+          echo "Version number updated from $current_version to $simple_version"
+
+      - name: Publish to nuget.org
+        run: | 
+          cd ./${{ inputs.working_dir }}
+          dotnet build UID2.Client.sln --configuration=Release
```

Comment:
1) We should run `dotnet test` and fail the pipeline if there are any test failures.
2) Do we need to build here? `dotnet pack` builds by default, so I think we're building twice.

Reply:
1. `dotnet test` sounds like a good idea, I will add that.
2. This is more a question for @jon8787 🥺 I followed the pattern in `BuildNuget.cmd`, any reason we want to build first?

Reply:
BuildNuget.cmd is years old so happy for you to change & test it :)

Reply:
Tested we can do `dotnet pack` without build, will update this step

Reply:
Sorry, thinking about this more - `dotnet test` also builds, so I think we want to `test` (in Release configuration) and then `pack` the build artifacts from `test`.